### PR TITLE
Add base image info labels to common package

### PR DESCRIPTION
**Pull request by jcfiorenzano**
**Task name**:  docker-common-packages-v2

**Description**:  Added methods to get the last base image name and the image digest.

**Documentation changes required:** N

**Added unit tests:** N

**Attached related issue:** N

**Checklist**:
- [ ] Task version was bumped - please check [instruction](https://github.com/microsoft/azure-pipelines-tasks/tree/master/docs/taskversionbumping.md) how to do it
- [ ] Checked that applied changes work as expected

### CODE REVIEW

File: common-npm-packages/docker-common-v2/containerimageutils.ts [hunk 1]
```diff
@@ -72,4 +73,125 @@ export function getResourceName(image: string, digest: string) {
...
+        for (var i = 0; i < lines.length; i++) {
+            var index = lines[i].toUpperCase().indexOf("FROM");
+            if (index == -1) {
+                continue;
+            }
+
+            var nameComponents = lines[i].substring(index + 4).toLowerCase().split(" as ");
+            var prospectImageName = nameComponents[0].trim();
+            if (nameComponents.length > 1) {
+                var tag = nameComponents[1].trim();
```
Comment (ajinkya599) [inline common-npm-packages/docker-common-v2/containerimageutils.ts:111]: Use names `alias` and `aliasToImageNameMapping` as `tag` has a different meaning in Docker context.

File: common-npm-packages/docker-common-v2/containerimageutils.ts [hunk 2]
```diff
@@ -72,4 +73,125 @@ export function getResourceName(image: string, digest: string) {
     namespace = namespace ? namespace + '/' : '';
     
     return "https://" + registry  + namespace  + repository + "@sha256:" + digest;
-  }
\ No newline at end of file
+  }
+
+export function getFinalBaseImageName(dockerFilePath: string): string {
```
Comment (ajinkya599) [inline common-npm-packages/docker-common-v2/containerimageutils.ts:78]: Continuing the discussion from:
https://github.com/microsoft/azure-pipelines-tasks/pull/14695#discussion_r626806057

Had a chat with @ds-ms. It makes sense to replace the existing `getBaseImageName` method as this is enhancing the logic.

**Review by ajinkya599 — COMMENTED** [review]
Bump-up the package version to `2.0.3`:

https://github.com/microsoft/azure-pipelines-tasks/blob/master/common-npm-packages/docker-common-v2/package.json#L3

File: common-npm-packages/docker-common-v2/containerimageutils.ts (hunk 2 above)
Comment (jcfiorenzano) [reply common-npm-packages/docker-common-v2/containerimageutils.ts]: Is there a way that we can test that we are not breaking other people scenarios. The expectations is that this new logic  should enhance the previous one but I would feel more confident if I see some kind of testing. I couldn't find a tests for the common packages functions.

Comment (ajinkya599) [reply common-npm-packages/docker-common-v2/containerimageutils.ts]: Good point. Having L0 tests would definitely have helped here. Unfortunately, its not there so we will need to rely on our own assessment.
It seems safe in the sense - It is guaranteed to give the result that the previous logic was giving or better.

**Review by ajinkya599 — APPROVED** [review]
🚀

**Pull request merged**